File: leetcode/MaximumSubarray/maxSubArray.java
public class Solution {
    public int maxSubArray(int[] A) {
        // Start typing your Java solution below
        // DO NOT write main() function
        int max = Integer.MIN_VALUE, cur = 0;
        for (int x : A) {
            cur += x;
            max = Math.max(max, cur);
            if (cur < 0) cur = 0;
        }
        return max;
    }
}

File: leetcode/MaximumSubarray/maxSubArray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        if (nums.size() < 1) return 0;
        int ans = nums[0], cur = 0;
        for (auto x : nums) {
            cur += x;
            ans = max(ans, cur);
            if (cur < 0) cur = 0;
        }
        return ans;
    }
};
